### PR TITLE
Have Jenkins test notebooks

### LINKED ISSUE #110: Automatically test/run tutorial notebooks
write tests for the correct workings of the tutorial  notebooks

### PR DESCRIPTION
**Pull request by hardbyte**
Uses the `nbval` plugin for pytest, runs inside a docker container on Jenkins.

Like clkhash only tests that none of the notebook cells error.

Closes #110

### CODE REVIEW

File: Jenkinsfile.groovy [hunk 1]
```diff
@@ -73,7 +77,8 @@ node('docker&&multicore&&ram') {
       try {
         echo("Start all the containers (including tests)")
         sh "docker-compose -v"
-          composeCmd(composeProject, "up -d db minio redis backend db_init worker nginx tests")
+        composeCmd(composeProject, "up -d db minio redis backend db_init worker nginx")
+        sleep 20
```
Comment (wilko77) [inline Jenkinsfile.groovy]: what are those sleep commands for? And is 20 the right value?

File: docs/tutorial/Similarity Scores.ipynb [hunk 2]
```diff
@@ -813,38 +893,69 @@
...
    ],
    "source": [
     "look_at_per_field_accuracy(threshold = 0.95, num_samples = 100)"
    ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [
```
Comment (wilko77) [inline docs/tutorial/Similarity Scores.ipynb]: what's that output for? There is no source for it...

File: Jenkinsfile.groovy (hunk 1 above)
Comment (hardbyte) [reply Jenkinsfile.groovy]: I lowered and moved the 30 second sleep from line 90 (before following the output of the integration test container). I suspect it gives a bit of buffer for pulling docker images.

File: Jenkinsfile.groovy [hunk 3]
```diff
@@ -82,12 +87,29 @@ node('docker&&multicore&&ram') {
       }
     }
 
+    stage('Tutorial Tests') {
+      //gitCommit.setInProgressStatus(gitContextDocumentation);
```
[the remark this answers is not in the document]
Comment (hardbyte) [reply Jenkinsfile.groovy]: Will do!

File: docs/tutorial/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Note we expect the user to mount the sources to build...
```
Comment (hardbyte) [inline docs/tutorial/Dockerfile]: this is incorrect. I'll remove....

**Pull request merged**